File: android/app/src/main/java/com/appxstudios/festivalconnection/models/ChannelInfo.kt
package com.appxstudios.festivalconnection.models

data class ChannelInfo(
    val id: String,
    var name: String,
    var creatorPublicKeyHex: String = "",
    var creatorDisplayName: String = "",
    var memberPublicKeys: MutableSet<String> = mutableSetOf(),
    var memberAvatarNames: MutableList<String> = mutableListOf(), // up to 4 display names
    var lastMessage: String = "",
    var lastMessageSenderName: String = "",
    var lastMessageTimestamp: Long? = null,
    var unreadCount: Int = 0,
    var isGeofenced: Boolean = true,
    var channelDescription: String = "",
    var createdAt: Long = System.currentTimeMillis(),
    var isVerified: Boolean = false,
    var isJoined: Boolean = false
) {
    val memberCount: Int get() = memberPublicKeys.size
}

File: ios/FestivalConnection/Models/ChannelInfo.swift
import Foundation

struct ChannelInfo: Identifiable, Equatable {
    let id: String
    var name: String
    var creatorPublicKeyHex: String = ""
    var creatorDisplayName: String = ""
    var memberPublicKeys: Set<String> = []
    var memberAvatarNames: [String] = [] // up to 4 display names for stacked avatars
    var lastMessage: String = ""
    var lastMessageSenderName: String = ""
    var lastMessageTimestamp: Date? = nil
    var unreadCount: Int = 0
    var isGeofenced: Bool = false
    var channelDescription: String = ""
    var createdAt: Date = Date()
    var isVerified: Bool = false
    var isJoined: Bool = false

    var memberCount: Int { memberPublicKeys.count }

    static func == (lhs: ChannelInfo, rhs: ChannelInfo) -> Bool {
        lhs.id == rhs.id
    }
}
